packages/custom_lint/lib/src/workspace.dart

```diff
@@ -633,39 +645,80 @@ publish_to: 'none'
               .nonNulls
               .toList(),
         ),
-    };
+    }.map(
+      (key, value) {
+        // Any dependency which is in both `dependencies` and `dev_dependencies` is treated as a regular dependency.
+        if (value.regularDependencies.isNotEmpty &&
+            value.devDependencies.isNotEmpty) {
+          value.regularDependencies.addAll(value.devDependencies);
+          value.devDependencies.clear();
+        }
+        return MapEntry(key, value);
+      },
+    );
 
-    // A flag for whether the "dependencies:" header has been written.
-    var didWriteDevDependenciesHeader = false;
-    // Write dev_dependencies
+    final dependencyConstraints = (
+      dependencies: <String, String>{},
+      devDependencies: <String, String>{},
+    );
+
+    // Iterate over each plugin and compute their constraints.
     for (final name in uniquePluginNames) {
       final allDependencies = dependenciesByName[name];
       if (allDependencies == null) continue;
 
-      // // We don't write dev_dependencies which are sometimes prod dependencies,
-      // // as dev_dependencies, because then we'd be specifying the dependency twice.
-      if (
-          // allDependencies.dependencies.isNotEmpty ||
-          allDependencies.devDependencies.isEmpty) {
+      // Determine if this is a dev_dependency or a dependency.
+      // Note: Dependencies can't be in both `dependencies` and `dev_dependencies`.
+      // So these isNonEmpty checks mutually exclusive.
+      final bool isDevDependency;
+      if (allDependencies.devDependencies.isNotEmpty) {
+        isDevDependency = true;
+      } else if (allDependencies.regularDependencies.isNotEmpty) {
+        isDevDependency = false;
+      } else {
+        // A dependency that is only in dependencyOverrides is handled separately below.
         continue;
       }
 
-      if (!didWriteDevDependenciesHeader) {
-        didWriteDevDependenciesHeader = true;
-        buffer.writeln('\ndev_dependencies:');
+      final String constraint;
+      if (allDependencies.dependencyOverrides.isNotEmpty) {
+        constraint = ' any';
```

Comment:
That's not right.

Reply:
Actually, it seems like that's how I handled it before....
It shoudn't be though :shrug: